packages/devtools_app/lib/src/flutter/split.dart

```diff
@@ -22,17 +22,22 @@ import 'package:flutter/material.dart';
 // TODO(djshuckerow): introduce support for a minimum fraction a child is allowed.
 class Split extends StatefulWidget {
   /// Builds a split oriented along [axis].
-  const Split({
+  Split({
     Key key,
     @required this.axis,
-    @required this.firstChild,
-    @required this.secondChild,
-    double initialFirstFraction,
-  })  : initialFirstFraction = initialFirstFraction ?? 0.5,
-        assert(axis != null),
-        assert(firstChild != null),
-        assert(secondChild != null),
-        super(key: key);
+    @required this.children,
+    @required this.initialFractions,
+  })  : assert(axis != null),
+        assert(children != null && children.length >= 2),
+        assert(initialFractions != null && initialFractions.length >= 2),
+        assert(children.length == initialFractions.length),
+        super(key: key) {
+    var sumFractions = 0.0;
+    for (var fraction in initialFractions) {
+      sumFractions += fraction;
+    }
+    assert(sumFractions == 1.0);
```

Comment:
need to watch for rounding error in this assert. Use an epsilon.

Reply:
done